internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,195 @@
+	matchLabels := make(map[string]string)
+	matchLabels["name"] = dsName
+	matchLabels["test-network-function.com/app"] = debug
+
+	var runAsPrivileged = true
+	var zeroInt int64
+	var zeroInt32 int32
+	var preempt = corev1.PreemptLowerPriority
+	var tolerationsSeconds int64 = 300
+	var hostPathType = corev1.HostPathDirectory
+
+	container := corev1.Container{
+		Name:            containerName,
+		Image:           imageWithVersion,
+		ImagePullPolicy: "IfNotPresent",
+		SecurityContext: &corev1.SecurityContext{
+			Privileged: &runAsPrivileged,
+			RunAsUser:  &zeroInt,
+		},
+		Stdin:                  true,
+		StdinOnce:              true,
+		TerminationMessagePath: "/dev/termination-log",
+		TTY:                    true,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				MountPath: "/host",
+				Name:      "host",
+			},
+		},
+	}
+	return &v1.DaemonSet{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        dsName,
+			Namespace:   namespace,
+			Annotations: dsAnnotations,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: matchLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       []corev1.Container{container},
+					PreemptionPolicy: &preempt,
+					Priority:         &zeroInt32,
+					HostNetwork:      true,
+					HostPID:          true,
+					Tolerations: []corev1.Toleration{
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/not-ready",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	logrus.Infof("Deleting daemon set %s", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy})
+	if err != nil {
+		return fmt.Errorf("daemonset %s deletion failed: %w", daemonSetName, err)
+	}
+
+	for start := time.Now(); time.Since(start) < timeout; {
+		client := clientsholder.GetClientsHolder().K8sClient
+		pods, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
```

Comment:
The label selector won't work for the current partner repo's daemonset. as it doesn't have the "name" label. Try using the debug label:
`LabelSelector: "test-network-function.com/app=debug"`
This should make the github's CI to pass.
You should also change this in line 179.